### ISSUE TITLE
Make sure video-manager-impl cleans up

### ISSUE THREAD

**Issue by sparhami** [ampproject/amphtml#17646]
The VideoManager listens for things and does not implement dispose. This means that if we create an instance for a shadow doc, and then later remove the shadow doc, some listeners will remain. For example: 

https://github.com/ampproject/amphtml/blob/36a08ab04326a2ec35c6d1ab7d19a86ab8d297d7/src/service/video-manager-impl.js#L1033

It should keep track and unlisten like layers:

https://github.com/ampproject/amphtml/blob/36a08ab04326a2ec35c6d1ab7d19a86ab8d297d7/src/service/layers-impl.js#L155-L159

**Comment by ampprojectbot** [comment]
This issue doesn't have a category which makes it harder for us to keep track of it.  Please add an appropriate category.

**Comment by ampprojectbot** [comment]
This issue doesn't have a category which makes it harder for us to keep track of it.  Please add an appropriate category.
